src/compiler/emitter.rs

```diff
@@ -0,0 +1,135 @@
+use std::cell::{RefCell, RefMut};
+use std::rc::Rc;
+
+use crate::{
+    is_keyword, token_to_string, Debug_, EmitHint, EmitTextWriter, LiteralTypeNode, Node,
+    NodeInterface, Printer, PrinterOptions, SourceFile, TypeNode,
+};
+
+#[derive(PartialEq, Eq)]
+enum PipelinePhase {
+    Notification,
+    Emit,
+}
+
+pub fn create_printer(printer_options: PrinterOptions) -> Printer {
```

Comment:
I guess the `typeToString()` function calls into the emitting code to "print" a type as a string so implemented a couple code paths for emitting

Reply:
Why does this live outside the `Printer` impl?

Reply:
I guess the pattern I've been using for these Typescript helpers that return an object with a bunch of closure "methods" is to keep the calling interface the same (so just calling a simple `create_printer()` function) but then have the Rust version create an instance of a struct type under the hood (so just the difference of calling as `create_printer()` being more similar to the Typescript version than eg `Printer::create()`)